### ISSUE TITLE
sf::Text::getLocalBounds() inconsistency

### ISSUE THREAD

**Issue by oomek** [SFML/SFML#2129]
I've recently realized that `sf::Text::getLocalBounds()` is a bit inconsistent.
When the text object has an outline applied then `top` and `left` return correctly compensated negative values, but `width` and `height` still contain the outline. This is a bit problematic to align the text to the right without it being affected by the outline.

**Comment by danieljpetersen** [comment]
Is this on all versions (2.5.1, 2.6, master)?

**Comment by oomek** [comment]
2.6.X

**Comment by eXpl0it3r** [comment]
Am I understanding this correctly that we subtract the outline width top-left, but don't add anything to the overall width-height of the local bounds?

The fact that the top-left is adjusted certainly makes sense to maintain a stable baseline. So the question essentially becomes whether the outline should affect the bounds size.

I currently can't think of a reason why it shouldn't. 🤔

Would this affect the line height, i.e. distance between lines?

**Comment by oomek** [comment]
I've prepared an example which will help you to understand what I mean.
When I try to align text to the right edge with fractional outline values the text is crawling.
Does not matter whether I align to outline or to text body.
With localBoundingBox() returning size without outline it would be cleaner. We could always add the outline value back instead of counting advance of outlined glyphs.

```cpp
#include <cmath>
#include <SFML/Window.hpp>
#include <SFML/Graphics.hpp>

int main(int argc, char *argv[])
{
    sf::RenderWindow window;
    window.create(sf::VideoMode(800, 600, 32), "SFML test");
    window.setVerticalSyncEnabled(true);
    window.setMouseCursorVisible(true);

    sf::Font font;
    std::vector<sf::Text> text;

    if(!font.loadFromFile("segoeui.ttf"))
        return EXIT_FAILURE;

    text.push_back(sf::Text("Lorem ipsum dolor sit amet,", font));
    text.push_back(sf::Text("consectetur adipiscing elit. Nam id", font));
    text.push_back(sf::Text("nisi id libero hendrerit eleifend. Sed", font));
    text.push_back(sf::Text("tincidunt vitae neque maximus", font));
    text.push_back(sf::Text("imperdiet.", font));

    int outline = 0;
    float lineHeight = 40;
    int alignToTextBody = 0;

    int pos = 0;
    for (auto& t : text)
    {
        t.setCharacterSize(27);
        t.setFillColor(sf::Color::White);
        t.setPosition(sf::Vector2f(100.0, 100.0 + pos));
        pos += lineHeight;
    }

    sf::RectangleShape rect1, rect2;
    rect1.setFillColor(sf::Color(0, 0, 0, 75));
    rect2.setFillColor(sf::Color(0, 0, 0, 75));
    rect1.setSize(sf::Vector2f(500.0, 200.0));
    rect2.setSize(sf::Vector2f(500.0, 200.0));
    rect1.setPosition(sf::Vector2f(150.0, 75.0));
    rect2.setPosition(sf::Vector2f(150.0, 325.0));

    while (window.isOpen())
    {
        sf::Event event;
        while (window.pollEvent(event))
        {
            if (event.type == sf::Event::Closed)
                window.close();
            if (event.type == sf::Event::KeyPressed)
            {
                // Switch alignment to text body, or to outline
                if (event.key.code == sf::Keyboard::Space)
                    alignToTextBody = 1 - alignToTextBody;

                // Increase text outline
                if (event.key.code == sf::Keyboard::Down)
                {
                    outline++;
                    for (auto& t : text) t.setOutlineThickness(outline * 0.2);
                }

                // Decrease text outline
                if (event.key.code == sf::Keyboard::Up)
                {
                    outline--;
                    if ( outline < 0 ) outline = 0;
                    for (auto& t : text) t.setOutlineThickness(outline * 0.2);
                }

                if (event.key.code == sf::Keyboard::Escape)
                {
                    window.close();
                    return EXIT_SUCCESS;
                }
            }
        }

        window.clear(sf::Color(100, 100, 200, 255));

        // Align left
        for ( int i = 0; i < text.size(); i++ )
        {
            float ol = ceilf(text[i].getOutlineThickness()) * alignToTextBody;
            text[i].setPosition(sf::Vector2f(rect1.getPosition().x - text[i].getLocalBounds().left - ol,
                                             rect1.getPosition().y + i * lineHeight - text[i].getLocalBounds().top - ol));
            window.draw(text[i]);
        }

        // Align right
        for ( int i = 0; i < text.size(); i++ )
        {
            float ol = ceilf(text[i].getOutlineThickness()) * alignToTextBody;
            text[i].setPosition(sf::Vector2f(rect2.getPosition().x + rect2.getSize().x - text[i].getLocalBounds().width - text[i].getLocalBounds().left + ol,
                                             rect2.getPosition().y + i * lineHeight - text[i].getLocalBounds().top - ol));
            window.draw(text[i]);
        }

        window.draw(rect1);
        window.draw(rect2);

        window.display();
    }
    return EXIT_SUCCESS;
}
```

**Comment by oomek** [comment]
Footnote:
Yes, I know I should not animate the outline ;)
I just want text positioning to be pixel perfect.

**Comment by eXpl0it3r** [comment]
Fixed with #2132
